staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -240,7 +240,8 @@ func (s *DelegatingAuthenticationOptions) getClientCA() (*ClientCertAuthenticati
 		return nil, err
 	}
 	if incluster == nil {
-		return nil, fmt.Errorf("cluster doesn't provide client-ca-file")
+		glog.Warningf("cluster doesn't provide client-ca-file in configmap/%s in %s, so client certificate authentication to extension api-server won't work.", authenticationConfigMapName, authenticationConfigMapNamespace)
```

Comment:
Are you suggesting a new flag like `SkipInClusterClientCaLookup` to not look up client-ca in ConfigMap when set to true?

If so, I have two follow up questions:
1. How do administrators reason between usage of existing `SkipInClusterLookup` and the new  `SkipInClusterClientCaLookup`? Won't that be confusing?
2. Isn't the [recommendation](https://github.com/kubernetes/kubernetes/issues/53990) to use an altogether separate CA for aggregator layer as a client? What are the scenarios in which extension API server needs to authenticate it's client using cluster's client CA?

Reply:
Sorry going to work on your questions out of order. 

The aggregation system uses two sets of certs, 1 to authenticate incoming requests from the Kube API Server (KAS) and another as the client cert it uses to identify itself in mutual TLS when making requests to the Kube API Server (and possibly other systems such as webhooks). The cert obtained from the config map is the cert used to authenticate the requester is the KAS. The reference recommendation has to do with if the second cert could be relied on as being signed by the same CA. Currently we allow it to be different and allow the the CA being given to the KAS via API registration. Then the aggregated API server would normally load this cert from a location on disk. In the sample api server we specify this location with the flags --tls-cert-file and --tls-private-key-file.

On your first question I am not sure there should be a difference. However we passed a check for SkipInClusterLookup, so we know its not set. However I do not like the idea of the admin having configured the system to use in cluster certs but we do something else because the client ca file was missing.